### DISCUSSION TITLE
Handling clicks on TrayIcon

### DISCUSSION THREAD

**Discussion by milkywayfarer** [AvaloniaUI/Avalonia#8297] [Q&A]
How to handle clicks (single and double) on TrayIcon? Setting `Clicked` event to literally anything in `App.xaml` file leads to `XAMLIL` compile error.

**Comment by maxkatz6** [comment]
You need to set a click event handler on NativeMenuItem.
What's the compiler error?

**Reply by milkywayfarer** [reply]
Is the TrayIcon in Windows System Tray actually a NativeMenuItem? I need to handle clicks directly on the icon in the tray without opening the menu.

Cannot get the exact compiler error right now, but I think it reads the x in <...Clicked="x"/> as a String type.

**Reply by maxkatz6** [reply]
From what I remember only menu is supported in the tray icon, without handling interaction with the icon itself. But I can be wrong here
 CC @danwalmsley

**Reply by maxkatz6** [reply]
> but I think it reads the x in <...Clicked="x"/> as a String type.

Normally this error is shown where compiler couldn't find a correct handler. And normally after that error it will print what's expected handler is, including parameter types.

**Reply by milkywayfarer** [reply]
After some fiddling with the code, I got the solution: putting a void with the following args: `object? sender, EventArgs e` is enough to compile the app and Clicked event is firing correctly. What type of EventArgs should I use specifically for the mouse? (MouseEventArgs depends on System.Windows.Forms, and adding that dependency loses the point of using Avalonia's TrayIcon)

**Reply by maxkatz6** [reply]
Just EventArgs. Also event documentation mentions it doesn't work on macOS. From what I remember it wasn't possible in AppKit api. 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/TrayIcon.cs#L79

**Reply by milkywayfarer** [reply]
Well, I'm writing an app that will be of no use on macOS platform anyway. Thanks for your replies!
